docs/content/preview/troubleshoot/nodes/disk-full.md

```diff
@@ -1,25 +1,136 @@
   preview:
     parent: troubleshoot-nodes
     weight: 40
 type: docs
 ---
 
-## Disk space full on crash loop
+This document describes the potential consequences and recommended actions when a YB-TServer or YB-Master encounters disk full conditions on the data drives.
 
-During a YB-TServer crash loop, the disk can fill up, preventing recovery.
+## Overview
 
-During the tablet bootstrap that takes place during a YB-TServer restart, a new WAL file is allocated, irrespective of the amount of logs generated in the server's previous WAL file. If a YB-TServer has several tablets and goes into a crash loop, each one of the tablets creates new WAL files on every bootstrap, filling up the disk.
+YugabyteDB data drive doesn't rely on the local root file system for storing data. Instead, YugabyteDB data directories are typically mounted on separate file systems to ensure isolation and manageability.
 
-Starting with v2.18.1, you can prevent repeated WAL file allocation in a crash loop by using the [--reuse_unclosed_segment_threshold_bytes](../../../reference/configuration/yb-tserver/#reuse-unclosed-segment-threshold-bytes) flag.
+The following flags are used to configure the data drive:
 
-If a tablet's last WAL file size is less than or equal to this threshold value, the bootstrap process will reuse the last WAL file rather than create a new one. To set the flag so that the last WAL file is always reused, you can set the flag to the current maximum WAL file size (64MB), as follows:
+- [fs_data_dirs](../../../reference/configuration/yb-tserver/#fs-data-dirs)
+- [log_dir](../../../reference/configuration/yb-tserver/#log-dir)
 
-```sh
-reuse_unclosed_segment_threshold_bytes=67108864
+If any of the data drives is approaching disk full condition, then the inserts, updates, or deletes that write to the tablet are rejected due to insufficient disk space, and they would fail with the following error:
+
+```output
+IO error (yb/tserver/tablet_service.cc:2288): Write to tablet 749e4d78244c43d2bba9cdb8505b732f rejected. Node 9a24d9c8493b44caa97ba4ae06e5829d has insufficient disk space
 ```
+
+The flags `max_disk_throughput_mbps` and `reject_writes_min_disk_space_mb` determine the amount of disk space that is considered too low. The default value is 3GB.
+
+The YB-TServer logs contain the following message when the system nears the threshold (18GB):
+
+```output
+W0829 15:35:51.325239 1986375680 log.cc:2186] Low disk space on yb-data/tserver/wals/table-7457ebcd745e4ea89375a30406f2c188/tablet-749e4d78244c43d2bba9cdb8505b732f/wal-000000001. Free space: 3518698209 bytes
+```
+
+and the following message when there is no more space left:
+
+```output
+0829 15:35:51.325239 1986375680 log.cc:2181] Not enough disk space available on yb-data/tserver/wals/table-7457ebcd745e4ea89375a30406f2c188/tablet-749e4d78244c43d2bba9cdb8505b732f/wal-000000001. Free space: 3518698209 bytes
+```
+
+The following options prevents YSQL/YCQL writes from consuming additional disk space:
```

Comment:
If the data drive is running short of disk space, use one of the following to stop the writes from failing.